app/components/course_preview/missing_information_component.rb

```diff
@@ -21,7 +21,11 @@ def link
     end
 
     def render?
-      FeatureService.enabled?(:course_preview_missing_information)
+      FeatureService.enabled?(:course_preview_missing_information) && preview?
+    end
+
+    def preview?
+      params[:action] == 'preview' || params[:action].nil?
```

Comment:
i don't think the components should reaching directly into the params, it should probably be passed in via an args and will make testing easier

Reply:
How do you mean?

Reply:
```ruby
def initialize(course:, information_type:, preview_mode: false)
      super
      @information_type = information_type
      @course = course
end

MissingInformationComponent.new(course:, information_type: :some_value, preview_mode: params[:action] == 'preview')

# or

MissingInformationComponent.new(course:, information_type: :some_value, action: params[:action])
```

Reply:
Oh right. Sorry, it was showing me your unedited comment for some reason 🤷🏼‍♂️. Will this do? 

https://github.com/DFE-Digital/publish-teacher-training/pull/3534/commits/c3f48c425a9ad332a85166d51bf23a16978848b6